rqd/rqd/rqcore.py

```diff
@@ -334,20 +335,15 @@ def runLinux(self):
 
         if not self.rqCore.updateRssThread.is_alive():
             self.rqCore.updateRssThread = threading.Timer(rqd.rqconstants.RSS_UPDATE_INTERVAL,
-                                                           self.rqCore.updateRss)
+                                                          self.rqCore.updateRss)
             self.rqCore.updateRssThread.start()
 
-        if rqd.rqconstants.RQD_PREPEND_TIMESTAMP:
-            pipe_to_file(frameInfo.forkedCommand.stdout, frameInfo.forkedCommand.stderr, self.rqlog)
-        else:
-            with open(self.rqlog.name, 'a') as f:
-                # Convert to ASCII while discarding characters that can not be encoded
-                for line in frameInfo.forkedCommand.stdout:
-                    line = line.encode('ascii', 'ignore')
-                    f.write(line.decode('ascii') + '\n')
-                for line in frameInfo.forkedCommand.stderr:
-                    line = line.encode('ascii', 'ignore')
-                    f.write(line.decode('ascii') + '\n')
+        while True:
+            output = frameInfo.forkedCommand.stdout.readline()
+            if not output and frameInfo.forkedCommand.poll() is not None:
+                break
+            if output:
+                self.rqlog.write(output, prependTimestamp=rqd.rqconstants.RQD_PREPEND_TIMESTAMP)
```

Comment:
This logic doesn't seem to have the same problems of the previous logic when `RQD_PREPEND_TIMESTAMP=False` (See #1426 ) but one think I'm not quite sure I understand is how stderr is being parsed.

The logic from `pipe_to_file` that got removed allows parsing both stdout and stderr into a single output without blocking on waiting on one of the pipes. Probably something similar will be required if we want both stdout and stderr content to be sent to Loki.

Reply:
From what I remember both, STDOUT and STDERR is being piped and thus end out the same way. I'm almost certain I tested exactly for that. Let me double check it.

Reply:
Yes, both STDOUT and STDERR works (tested it with the file output, but should be the same for loki)

Reply:
Btw, will it be ok to have the loki part still being beta/experimental? This PR changes alot more than just adding support for loki.

Reply:
Now I remember how it was done. stderr is sent to stdout, of course there won't be any distinguish between the 2, which might be good to have in loki as seperate tags.
```python
stderr=subprocess.STDOUT,
```